cpp/src/arrow/extension_type.h

```diff
@@ -80,6 +81,12 @@ class ARROW_EXPORT ExtensionType : public DataType {
   /// \return the serialized representation
   virtual std::string Serialize() const = 0;
 
+  static std::shared_ptr<Array> WrapArray(const std::shared_ptr<DataType>& ext_type,
+                                          const std::shared_ptr<Array>& storage);
+  static std::shared_ptr<ChunkedArray> WrapArray(
+      const std::shared_ptr<DataType>& ext_type,
+      const std::shared_ptr<ChunkedArray>& storage);
```

Comment:
@wesm Thoughts about this API?

Reply:
Seems OK to me. May want doxygen comments

Reply:
Right :-)